shiny/ui/_theme.py

```diff
@@ -0,0 +1,280 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Literal, Optional
+from warnings import warn
+
+from htmltools import HTMLDependency, Tag, Tagifiable, TagList
+from htmltools.tags import head
+from packaging.version import Version
+
+from .._docstring import add_example
+from .._versions import bootstrap as BOOTSTRAP_VERSION
+from ._include_helpers import include_css
+
+__all__ = ("theme",)
```

Comment:
I feel uneasy about having an exported `shiny.ui.theme()` that what is mainly a advanced user / developer facing API.

Also, I can't help but feel like `Theme`/`theme` is a bit of a misnomer, especially if we're thinking about this as a user facing API. In that case, maybe something like `bootstrap_css()` is better?

In fact, I hate to suggest this, but maybe we don't need a `theme` argument at all for this use case. Could we not have a `bootstrap_css()` that could be dropped anywhere in the UI? Maybe done something like this?

```python
__all__ = (
    "bootstrap_css",
)

def bootstrap_css(path: Path | str, check_version: bool = True) -> list[Tag | HTMLDependency]:
    # Snoop the version of the Bootstrap CSS file
    if check_version:
        check_bootstrap_version(path)

    return [
        ui.include_css(path),
        HTMLDependency(
            name="bootstrap-css", version=bootstrap_version
        ),
    ]

def check_bootstrap_version(path: Path | str):
    file_path = check_path(path)
    content = read_utf8(file_path)
    # TODO: extract version from Bootstrap's banner, then warn/err if not compatible, and
    # also warn if the version wasn't found?

def bootstrap_deps() -> list[HTMLDependency]:
    return [
        jquery_deps(),
        HTMLDependency(
            name="bootstrap-js",
            version=bootstrap_version,
            source={"package": "shiny", "subdir": "www/shared/bootstrap/"},
            script={"src": "bootstrap.bundle.min.js"},
            meta={"name": "viewport", "content": "width=device-width, initial-scale=1"},
        ),
        HTMLDependency(
            name="bootstrap-css",
            version=bootstrap_version,
            source={"package": "shiny", "subdir": "www/shared/bootstrap/"},
            stylesheet={"href": "bootstrap.min.css"}
        )
    ]
```

Reply:
Two things to note about this approach:

1. We'd (attempt to) check the bootstrap version automatically. As long as Bootstrap isn't getting compiled in a weird way, this should work.

2. We'd (intentionally) drop the `replace = "all"` feature, but we could bring it back with a `bootstrap_js()` if need be down the road. In general though, the thought of encouraging users to customize the JS gives me the heebie jeebies.

Reply:
Oh sorry, just saw your comment, yes will do.